test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,147 @@
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static BufferedImage displayImage;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::initializeGUI);
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("ScreenCaptureRobotTest Frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        robot.delay(delay);
+        robot.waitForIdle();
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
+
+        // Capturing Image using Robot
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image";
+            System.err.println("Test failed");
+            throw new RuntimeException(errorMessage);
+        } else {
+            System.out.println("\nCaptured Image is same as actual Image");
+            System.out.println("Test passed");
+        }
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) {
+        int capturedPixel;
+        int realPixel;
+        int imgWidth;
+        int imgHeight;
+        boolean result = true;
+
+        imgWidth = capturedImg.getWidth(null);
+        imgHeight = capturedImg.getHeight(null);
+
+        for (int i = 0; i < (imgWidth); i++) {
+            for (int j = 0; j < (imgHeight); j++) {
+                capturedPixel = capturedImg.getRGB(i, j);
+                realPixel = realImg.getRGB(i, j);
+                if (capturedPixel != realPixel) {
+                    result = false;
+                    return result;
+                }
+            }
+        }
+        return result;
+    }
+
```

Comment:
```suggestion
//Save BufferedImage to PNG file
    private static void saveImage(BufferedImage image, String fileName) {
        if (image != null) {
            try {
                File file = new File(fileName);
                System.out.println("Saving button image to " + file.getAbsolutePath());
                ImageIO.write(image, "PNG", file);
            } catch (Exception e) {
                throw new RuntimeException("Could not write image file");
            }
        } else {
            throw new RuntimeException("BufferedImage was set to null");
        }
    }
```

Reply:
Yes, please save the images, especially when comparison fails.

I think handling `image != null` is not needed here. If `null` is passed, the test will fail with `NullPointerException`, and it's acceptable. Without the `if` statement, the code to save the image would be cleaner.

Reply:
saved images without null check.